src/snowflake/cli/plugins/nativeapp/manager.py

```diff
@@ -254,6 +257,26 @@ def app_post_deploy_hooks(self) -> Optional[List[ApplicationPostDeployHook]]:
     def debug_mode(self) -> bool:
         return self.na_project.debug_mode
 
+    @contextmanager
+    def use_warehouse(self, new_wh: Optional[str]):
+        """
+        Switches to a different warehouse for a while, then switches back.
+        This is a no-op if the requested warehouse is already active, or the requested warehouse is not available.
+        """
+        wh_result = self._execute_query(
+            f"select current_warehouse()", cursor_class=DictCursor
+        ).fetchone()
+        prev_wh = wh_result["CURRENT_WAREHOUSE()"]
+        is_different_wh = (new_wh is not None) and (new_wh.lower() != prev_wh.lower())
+        if is_different_wh:
+            self._log.debug("Assuming different warehouse: %s", new_wh)
```

Comment:
nit: maybe something like "Switching to warehouse x" or "Assuming warehouse x"... 'different' seems a bit off.

Reply:
Also similar to the `use_role` contextmanager. If we change it here, we'd have to change it there too. I can add that in for both of them. Just FYI, that will also require SnowCLI team's approval, so it might introduce a bit of a delay with their business hours.